### PR TITLE
[NTUSER] Check class name on IntGetAtomFromStringOrAtom

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Based on `I_Kill_Bugs`' `class_error.patch`.
JIRA issue: [CORE-13422](https://jira.reactos.org/browse/CORE-13422), [CORE-7531](https://jira.reactos.org/browse/CORE-7531), [CORE-18639](https://jira.reactos.org/browse/CORE-18639)

## TODO

- [x] Do tests.

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]
Approved. Please wait for approval of @hbelusca before merging. He had an open question in the ticket regarding this patch.

File: win32ss/user/ntuser/class.c
Comment (HBelusca) [inline win32ss/user/ntuser/class.c:1366]: This is an error code that has to be set _**by the caller of this helper function**_
Indeed, even though the string parameter is called "class" name, the function is also used with other strings that are _**NOT**_ describing window classes.

**Review by HBelusca — CHANGES_REQUESTED** [review]

File: win32ss/user/ntuser/class.c
Comment (katahiromz) [reply win32ss/user/ntuser/class.c:1366]: > the caller of this helper function

The caller of this helper function is plural...

**Comment by katahiromz** [comment]
> @katahiromz Could you check whether this PR also fixes the two applications mentioned in this JIRA report: https://jira.reactos.org/browse/CORE-7531 ? If so, please also add this JIRA report number to the commit log.
> 
> EDIT: Also this report https://jira.reactos.org/browse/CORE-18639 should be mentioned in the commit.

KRosUser did test successfully on both.

**Review by HBelusca — APPROVED** [review]
@katahiromz You can commit the PR. I'll submit another one later to try to make that function slightly more general later.

**Pull request merged**
